src/games/strategy/triplea/delegate/BattleDelegateOrdered.java

```diff
@@ -0,0 +1,114 @@
+      }
+    }
+  }
+
+  private int fightAmphib( IBattle amphib ) {
+    int foughtBattleCount = 0;
+    // are there battles that must occur first
+    for( final IBattle seaBattle : m_battleTracker.getDependentOn( amphib ) ) {
+      if( seaBattle instanceof MustFightBattle && seaBattle != null ) {
+        seaBattle.fight( m_bridge );
+        foughtBattleCount++;
+      }
+    }
+    amphib.fight( m_bridge );
+    foughtBattleCount++;
+    
+    return foughtBattleCount;
+  }
+
+  private int processAmphib() {
+    // Fight all amphibious assaults with no retreat option for the attacker and no sea combat - these have no real attacker decisions
+    // Also remove all remaining defenseless fights by fighting them
+    int battleCount = 0;
+    int amphibCount = 0;
+    IBattle lastAmphib = null;
+    for( final Territory t : m_battleTracker.getPendingBattleSites(false) ) {  // Loop through normal combats i.e. not bombing or air raid
+      final IBattle battle = m_battleTracker.getPendingBattle(t, false, BattleType.NORMAL);
+      if( battle instanceof NonFightingBattle && m_battleTracker.getDependentOn(battle).isEmpty() ) {
+        battle.fight( m_bridge );         // Remove non fighting battles by fighting them automatically. Conveniently done here.
+        continue;
+      } else if (!(battle instanceof MustFightBattle) ) {
+        continue;
+      }
+      battleCount++;
+      if( battle.isAmphibious() ) {
+        // If there is no dependent sea battle and no retreat, fight it now 
+        if( m_battleTracker.getDependentOn(battle).isEmpty() && !( (MustFightBattle) battle).canAnyAttackersRetreat() ) {
+          battle.fight( m_bridge );
+        } else {
+          amphibCount++;
+          lastAmphib = battle;      // Otherwise store it to see if it can be fought later i.e. if there's only one
+        }
+      }
+    }
+
+    // Fight amphibious assault if there is one remaining. Fight naval battles in random order if prerequisites first.
+    if( amphibCount == 1 ) {
+      battleCount -= fightAmphib( lastAmphib );   // Subtract off the number of fought battles 
+    }
+    return battleCount;
+  }
+
+  private void fightRemainingBattle() {
+    // This needs another loop just in case the last otherBattleCount was triggered by a sea combat already fought
+    for( final Territory t : m_battleTracker.getPendingBattleSites(false) ) {  // Will only find one
+      final IBattle lastBattle = m_battleTracker.getPendingBattle( t, false, BattleType.NORMAL );
+      if( lastBattle instanceof MustFightBattle ) {
+        lastBattle.fight( m_bridge );
+      }
+    }
+  }
+
+  /**
+   * After the super class .start() method runs, this is called to fight all battles not requiring significant attacker input,
+   * in the order required for Global 1940. Other maps with the same required order can also use it or add it anyway
+   */
+  @Override
+  public void start() {
+    super.start();
+
+    stratBombing();
```

Comment:
> I don't follow your point in almost every word of that.

Not helpful, be specific, ask questions.

> Break it up more? It is already a pile of spaghetti. Was reasonably straight forward the way I originally had it.

You arbitrarily split the lines of code up into different functions. This is not abstracting higher order logic away from implementation details. Did you expect this to be a whole lot better? The ordering in which things are done is now clear, but there are dependencies on those orderings that should not be there, and it is your job to not build a system that relies on it. Such a system is brittle.

>  getPendingBattleSites and getPendingBattle methods to get rid of the boolean parameter? That would require either duplication of the method or changing all the existing calls to it. 

No, create an API which you like, use that, have this new API forwards calls to the old one. Then only use the new API. Deprecate the original method if we should stop calling it completely in favor of the new API.

If not, that is the concern.


@simon33-2  - I think you are missing many points. I appreciate you trying to humor me in areas, but it's not very helpful when the actual goals are missed.  Build a reasonable unit test for this code that covers it effectively. Whatever else you do, I really don't much care so long as it takes about less than 5 minutes ballpark to understand this module, ideally more like 2 or 3 minutes given it is only 100 lines.